httphandlers/http_handlers_test.go

```diff
@@ -147,7 +148,9 @@ func TestWriteLog(t *testing.T) {
 				"allowed-header": {"test-header"},
 				"denied-header":  {"ignore-key"},
 			},
-			deniedHeaders: []string{"denied-header"},
+			deniedHeaders: func(key string) bool {
```

Comment:
The simple case of denying single header looks ok, but in general the user might need to filter multiple headers. That will mean that she will need to provide the headers as well as the logic for filtering them in `deniedHeaders`. This means more work for the user and library that is not so useful. In my opinion the previous version (with collection of values to be filtered) was simpler and easier to use.

Reply:
For the simple cases collection of values would do, but some times we don't know or don't want to spell out all the all the headers. So my initial thought was to request a collection of regexp from the user. But then decided that is cleaner to leave it to the user to choose how to filter.

Reply:
Yes, it is cleaner for the library but my point is that it not as clean for the user. Could we provide more test cases as documentation about using different filter functions? Or instead of test cases may be some [examples](https://blog.golang.org/examples)?